software/lofreq/callparallel/main.nf

```diff
@@ -0,0 +1,43 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process LOFREQ_CALLPARALLEL {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::lofreq=2.1.5" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/lofreq:2.1.5--py38h588ecb2_4"
+    } else {
+        container "quay.io/biocontainers/lofreq:2.1.5--py38h588ecb2_4"
+    }
+
+    input:
+    tuple val(meta), path(bam)
+    file bam_index
+    file fasta
+    file fasta_index
```

Comment:
```suggestion
    tuple val(meta), path(bam), path(bai)
    file  fasta
    file  fai
```

Reply:
The `bam` and `bai` are normally shipped together for each sample so should be defined in the same channel. Does `lofreq` fail without the BAM index and FASTA index?

Reply:
Yes, lofreq fails without the BAM index and FASTA index. I have now defined the bam and bai in the same channel and it is working fine.